lib/posix-fdio/fdctl.c

```diff
@@ -38,6 +40,10 @@ int uk_sys_ioctl(struct uk_ofile *of, int cmd, void *arg)
 	r = uk_file_ctl(f, UKFILE_CTL_IOCTL, cmd, (uintptr_t)arg, 0, 0);
 	if (iolock)
 		uk_file_wunlock(f);
+
+	/* Universally return -ENOTTY for unimplemented TTY ioctls */
+	if (unlikely(r == -ENOSYS && IOCTL_CMD_ISTYPE(cmd, IOCTL_CMD_TYPE_TTY)))
```

Comment:
I assume `r` here can only be `ENOSYS` in case the second condition is true, correct?

Reply:
Not really, even a "true" tty might legitimately error out of these ioctls because of e.g. bad arguments or whatever. So stuff like `-EINVAL` and `-EFAULT` and the sort are very much possible, and we purposely want to propagate those errors out to userspace.

Reply:
I see, makes sense. Just wanted to make sure :pray:.